modules/system/combat.js

```diff
@@ -14,47 +14,50 @@ export default class CombatHelpers {
         endCombatScripts: [CombatHelpers.checkCorruption, CombatHelpers.checkInfection, CombatHelpers.checkDiseases]
     }
 
-    static combatChecks(combat, type) {
-        CombatHelpers.scripts[type].forEach(script => { script(combat) })
+    static async combatChecks(combat, type) {
+        let scripts = CombatHelpers.scripts[type];
+        for (let script of scripts) {
+           await script(combat);
+        }
     }
 
-    static preUpdateCombat(combat, updateData) {
+    static async preUpdateCombat(combat, updateData) {
         if (!updateData.round && !updateData.turn)
             return
         if (combat.round == 0  && combat.active) {
-            CombatHelpers.combatChecks(combat, "startCombat")
+            await CombatHelpers.combatChecks(combat, "startCombat")
         }
         if (combat.round != 0 && combat.turns && combat.active) {
             if (combat.current.turn > -1 && combat.current.turn == combat.turns.length - 1) {
-                CombatHelpers.combatChecks(combat, "endRound")
+                await CombatHelpers.combatChecks(combat, "endRound")
             }
         }
 
-        CombatHelpers.combatChecks(combat, "endTurn")
+        await CombatHelpers.combatChecks(combat, "endTurn")
     }
 
-    static updateCombat(combat, updateData) {
+    static async updateCombat(combat, updateData) {
         if (!updateData.round && !updateData.turn)
             return
         if (combat.round != 0 && combat.turns && combat.active) {
-            CombatHelpers.combatChecks(combat, "startTurn")
+            await CombatHelpers.combatChecks(combat, "startTurn")
         }
     }
 
-    static endCombat(combat) {
-        CombatHelpers.combatChecks(combat, "endCombat")
+    static async endCombat(combat) {
+        await CombatHelpers.combatChecks(combat, "endCombat")
```

Comment:
These probably don't need to be async. Even if they call `runEffectsAsync` doesn't mean you need to await it